platform/android/src/style/sources/geojson_source.cpp

```diff
@@ -175,25 +173,23 @@ namespace android {
 
     void GeoJSONSource::setAsync(Update::Converter converterFn) {
         awaitingUpdate = std::make_unique<Update>(
-                std::move(converterFn),
-                std::make_unique<Actor<Callback>>(
-                        *Scheduler::GetCurrent(),
-                        [this](GeoJSON geoJSON) {
-                            // conversion from Java features to core ones finished
-                            android::UniqueEnv _env = android::AttachEnv();
-
-                            // Update the core source
-                            source.as<mbgl::style::GeoJSONSource>()->GeoJSONSource::setGeoJSON(geoJSON);
-
-                            // if there is an awaiting update, execute it, otherwise, release resources
-                            if (awaitingUpdate) {
-                                update = std::move(awaitingUpdate);
-                                update->converterFn(update->callback->self());
-                            } else {
-                                update.reset();
-                            }
-                        })
-        );
+            std::move(converterFn),
+            std::make_unique<Actor<GeoJSONDataCallback>>(
+                *Scheduler::GetCurrent(), [this](std::shared_ptr<style::GeoJSONData> geoJSONData) {
+                    // conversion from Java features to core ones finished
+                    android::UniqueEnv _env = android::AttachEnv();
+
+                    // Update the core source
+                    source.as<mbgl::style::GeoJSONSource>()->setGeoJSONData(std::move(geoJSONData));
```

Comment:
Does `setGeoJSONData` operate in a fixed-order manner?

Reply:
The intended change does not modify the existing coalescing logic, setGeoJSONData is just provided with the ready-to-use tiles